From dc3f73994646583ca56301a6a0cab9d0eb0435f6 Mon Sep 17 00:00:00 2001
Message-Id: <dc3f73994646583ca56301a6a0cab9d0eb0435f6.1612979959.git.marksvc@gmail.com>
From: MarkS <marksvc@gmail.com>
Date: Wed, 10 Feb 2021 11:32:33 -0600
Subject: [PATCH] flatpak environ

- Detect when running in flatpak and set environment variables
accordingly.
- Find more things in INSTALLATION_PREFIX, not simply /usr.

Change-Id: I17f91f6edc5524993d86b875d3e1db3bb6de407a
---
 environ | 22 +++++++++++++++-------
 1 file changed, 15 insertions(+), 7 deletions(-)

diff --git a/environ b/environ
index ae095247..26373069 100644
--- a/fw/environ
+++ b/fw/environ
@@ -30,6 +30,15 @@ if [ -f "${DOTNET_TOOLS_PATHER}" ]; then
   . "${DOTNET_TOOLS_PATHER}"
 fi
 
+if [[ -d /app/lib/fieldworks ]]; then
+  # We are running in flatpak.
+  MONO_PREFIX=/app
+  MONO_SILPKGDIR=/app
+  INSTALLATION_PREFIX=/app
+  LD_LIBRARY_PATH="/app/lib:${LD_LIBRARY_PATH}"
+  PATH="/app/bin:${PATH}"
+fi
+
 BASE=$(pwd)
 INSTALLATION_PREFIX="${INSTALLATION_PREFIX:-/usr}"
 COM=$(dirname "${BASE}")/libcom/COM
@@ -42,7 +51,7 @@ MONO_SILPKGDIR="${MONO_SILPKGDIR:-/opt/mono5-sil}"
 # search for xulrunner and geckofx, select the best, and add its location to LD_LIBRARY_PATH
 . ./environ-xulrunner
 [ -d /usr/lib/cli/gdk-sharp-2.0 ] && GDK_SHARP=/usr/lib/cli/gdk-sharp-2.0/
-ENC_CONVERTERS=/usr/lib/fieldworks
+ENC_CONVERTERS="${INSTALLATION_PREFIX}/lib/fieldworks"
 
 MONO_RUNTIME=v4.0.30319
 MONO_DEBUG=explicit-null-checks
@@ -52,13 +61,12 @@ MONO_ENV_OPTIONS="-O=-gshared"
 if [ "$RUNMODE" != "INSTALLED" ]
 then
 	PATH="${BASE}/Output_${ARCH}/${BUILD}:\
-/usr/lib/fieldworks/icu-bin:\
+${INSTALLATION_PREFIX}/lib/fieldworks/icu-bin:\
 ${COM}/build${ARCH}/bin:\
 ${PATH}"
 	LD_LIBRARY_PATH="${BASE}/Output_${ARCH}/${BUILD}:\
-/usr/lib/fieldworks/lib:\
+${INSTALLATION_PREFIX}/lib/fieldworks/lib:\
 ${COM}/build${ARCH}/lib:\
-/app/lib:\
 ${LD_LIBRARY_PATH}"
 fi
 # ensure we scan the default pkg-config directories (to pick up Geckofx for compiling)
@@ -67,7 +75,7 @@ PKG_CONFIG_PATH="${PKG_CONFIG_PATH:-/usr/lib/pkgconfig:/usr/share/pkgconfig}"
 # Add packaged mono items to paths
 PATH="${MONO_SILPKGDIR}/bin:${PATH}"
 LD_LIBRARY_PATH="${MONO_SILPKGDIR}/lib:${ENC_CONVERTERS}:${LD_LIBRARY_PATH}"
-PKG_CONFIG_PATH="${MONO_SILPKGDIR}/lib/pkgconfig:/usr/lib/fieldworks/lib/pkgconfig:${PKG_CONFIG_PATH}"
+PKG_CONFIG_PATH="${MONO_SILPKGDIR}/lib/pkgconfig:${INSTALLATION_PREFIX}/lib/fieldworks/lib/pkgconfig:${PKG_CONFIG_PATH}"
 MONO_GAC_PREFIX="${MONO_SILPKGDIR}:${ENC_CONVERTERS}:/usr:${MONO_PREFIX}"
 
 if [ "$RUNMODE" != "PACKAGING" ]
@@ -118,9 +126,9 @@ MONO_MWF_SCALING=disable
 # If the standard installation directory for FLExBridge exists, and the environment
 # variable is not yet set, set the environment variable for finding FLExBridge.
 # (Setting the LocalMachine registry value at installation doesn't work for Linux.)
-if [ -z "$FLEXBRIDGEDIR" -a -d "/usr/lib/flexbridge" ]
+if [ -z "$FLEXBRIDGEDIR" -a -d "${INSTALLATION_PREFIX}/lib/flexbridge" ]
 then
-	FLEXBRIDGEDIR="/usr/lib/flexbridge"
+	FLEXBRIDGEDIR="${INSTALLATION_PREFIX}/lib/flexbridge"
 fi
 
 export \
-- 
2.25.1

